File: Frontend/src/Pages/Profiles/StudentProfile.js
import axios from "axios";
import { useEffect, useState } from "react";
import { useHistory, useLocation } from "react-router"

const StudentProfile = () => {

    const location = useLocation()
    const data = location.state
    const history = useHistory();
    useEffect(() => {
        Defaults()
    }, [])

    const [userId, setUserId] = useState(data.userId);
    const [firstName, setFirstName] = useState(data.firstName);
    const [lastName, setLastName] = useState(data.lastName)
    const [dob, setDob] = useState(data.dob)
    const [gender, setGender] = useState(data.gender)
    const [locality, setLocality] = useState(data.locality)
    const [mobile, setMobile] = useState(data.mobile)
    const [pincode, setPincode] = useState(data.pincode)
    const [section, setSection] = useState(data.section)
    const [std, setStd] = useState(data.std)
    const [username, setUsername] = useState(data.username)

    function Defaults() {
        //console.log(data.firstName)
        document.getElementById("firstName").defaultValue = data.firstName
        document.getElementById("lastName").defaultValue = data.lastName
        document.getElementById("gender").defaultValue = data.gender
        document.getElementById("dob").defaultValue = data.dob
        document.getElementById("locality").defaultValue = data.locality
        document.getElementById("mobile").defaultValue = data.mobile
        document.getElementById("pincode").defaultValue = data.pincode
        document.getElementById("section").defaultValue = data.section
        document.getElementById("className").defaultValue = data.std
        document.getElementById("username").defaultValue = data.username
    }

    const UpdateProfile = () => {
        //console.log(data)
        const body = {
            "userId": userId,
            "firstName": firstName,
            "lastName": lastName,
            "dob": dob,
            "gender": gender,
            "locality": locality,
            "mobile": mobile,
            "pincode": pincode,
            "section": section,
            "std": std,
            "username": username
        }
        const url = "http://localhost:8080"
        axios.post(url + "/student/updateProfile", body).then(Response => {
            const result = Response.data
            if (result.status == "success") {
                window.alert("User Updated successfully")
                history.push('/AdminHome')
            } else {
                window.alert(result.error)
            }
        })
    }

    return (
        <div className="container">
            <div className="adminProfile"><h4 asign="center"> Student Profile</h4>
                <table>
                    <tbody>
                        <tr>
                            <td>Registration Id : </td>
                            <td>
                                <input type="number" disabled value={data.userId} onChange={e => { setUserId(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>First Name : </td>
                            <td>
                                <input id="firstName" type="text" onChange={e => { setFirstName(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Last Name : </td>
                            <td>
                                <input id="lastName" type="text" onChange={e => { setLastName(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Date of Birth : </td>
                            <td>
                                <input id="dob" type="date" onChange={e => { setDob(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Gender : </td>
                            <td>
                                <div className="input-box">

                                    <select id="gender" name="gender" onChange={e => { setGender(e.target.value) }}>
                                        <option value={gender}>{gender}</option>
                                        <option value="male">Male</option>
                                        <option value="female">Female</option>
                                        <option value="other">Other</option>
                                    </select>
                                </div>

                            </td>
                        </tr>
                        <tr>
                            <td>Locality : </td>
                            <td>
                                <input id="locality" type="text" onChange={e => { setLocality(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Mobile : </td>
                            <td>
                                <input id="mobile" type="number" onChange={e => { setMobile(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Pincode : </td>
                            <td>
                                <input id="pincode" type="number" onChange={e => { setPincode(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Section : </td>
                            <td>
                                <input id="section" disabled type="text" onChange={e => { setSection(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Class : </td>
                            <td>
                                <input id="className" disabled type="text" onChange={e => { setStd(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td>Username : </td>
                            <td>
                                <input id="username" type="text" onChange={e => { setUsername(e.target.value) }} />
                            </td>
                        </tr>
                        <tr>
                            <td></td>
                            <td colSpan="2">
                                <button className="button-show" onClick={UpdateProfile} >Update</button>
                            </td>

                        </tr>
                    </tbody>
                </table>
            </div></div>
    )
}
export default StudentProfile

File: sms1/src/main/java/com/iacsd/dto/StudentProfile.java
package com.iacsd.dto;

import java.util.Date;

public class StudentProfile {
	private int userId;
	private String firstName;
	private String lastName;
	private Date dob;
	private String mobile;
	private String gender;
	private String username;
	private String pincode;
	private String locality;
	private String std;
	private String section;
	private String typeJob;
	
	public StudentProfile() {
	}

	

	public StudentProfile(int userId, String firstName, String lastName, Date dob, String mobile, String gender,
			String username, String pincode, String locality, String std, String section, String typeJob) {
		this.userId = userId;
		this.firstName = firstName;
		this.lastName = lastName;
		this.dob = dob;
		this.mobile = mobile;
		this.gender = gender;
		this.username = username;
		this.pincode = pincode;
		this.locality = locality;
		this.std = std;
		this.section = section;
		this.typeJob = typeJob;
	}



	public String getTypeJob() {
		return typeJob;
	}



	public void setTypeJob(String typeJob) {
		this.typeJob = typeJob;
	}



	public int getUserId() {
		return userId;
	}

	public void setUserId(int userId) {
		this.userId = userId;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public Date getDob() {
		return dob;
	}

	public void setDob(Date dob) {
		this.dob = dob;
	}

	public String getMobile() {
		return mobile;
	}

	public void setMobile(String mobile) {
		this.mobile = mobile;
	}

	public String getGender() {
		return gender;
	}

	public void setGender(String gender) {
		this.gender = gender;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	public String getPincode() {
		return pincode;
	}

	public void setPincode(String pincode) {
		this.pincode = pincode;
	}

	public String getLocality() {
		return locality;
	}

	public void setLocality(String locality) {
		this.locality = locality;
	}

	public String getStd() {
		return std;
	}

	public void setStd(String std) {
		this.std = std;
	}

	public String getSection() {
		return section;
	}

	public void setSection(String section) {
		this.section = section;
	}

	@Override
	public String toString() {
		return "StudentProfile [userId=" + userId + ", firstName=" + firstName + ", lastName=" + lastName + ", dob="
				+ dob + ", mobile=" + mobile + ", gender=" + gender + ", username=" + username + ", pincode=" + pincode
				+ ", locality=" + locality + ", std=" + std + ", section=" + section + "]";
	}
}
